src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -0,0 +1,178 @@
+)
+object Notebot : Module() {
+
+    private var firstNote: Long = 0L
+    private var elapsed: Long = 0L
+    var playingSong = false
+
+    private var noteSequence: TreeMap<Long, ArrayList<Note>> = TreeMap()
+    var instrumentMap = InstrumentMap()
+
+    var channelMap: TreeMap<Int, Instrument> = TreeMap()
+
+    /* This is a bad way to have channels. In the future, I would like for there to be some way to display a map.*/
+    @Setting(name = "Channel 0")
+    var channelZero = Instrument.HAT
+
+    @Setting(name = "Channel 1")
+    var channelOne = Instrument.HAT
+
+    @Setting(name = "Channel 2")
+    var channelTwo = Instrument.HAT
+
+    @Setting(name = "Channel 3")
+    var channelThree = Instrument.HAT
+
+    @Setting(name = "Channel 4")
+    var channelFour = Instrument.HAT
+
+    @Setting
+    var hatAlwaysAsFSharp = true
+
+    @Setting(comment = "Name of MIDI file in the songs folder of your .minecraft directory")
+    @ImGuiExtra.Post("renderButtons")
+    var songName = "tetris_theme.mid"
+
+    fun renderButtons() {
+        dsl.button("Discover") { discoverSong() }
+        dsl.button("Load") { loadSong() }
+        playButton()
+        pauseButton()
+    }
+
+    private fun playButton() {
+        wrapDisabled(playingSong) {
+            dsl.button("Play") {
+                firstNote = System.currentTimeMillis() - elapsed
+                playingSong = true
+            }
+        }
+    }
+
+    private fun pauseButton() {
+        wrapDisabled(!playingSong) {
+            dsl.button("Pause") {
+                playingSong = false
+            }
+        }
+    }
+
+    private fun discoverSong() {
+        val player = mc.player ?: return
+        for (x in -4..4) {
+            for (y in -4..4) {
+                for (z in -4..4) {
+                    val pos = player.pos?.plus(Vec3d(x.toDouble(), y.toDouble(), z.toDouble())) ?: break
+                    if (player.world.getBlockState(BlockPos(pos))?.block == Blocks.NOTE_BLOCK) {
+                        val noteBlock = player.world.getBlockState(BlockPos(pos))
+                        instrumentMap.add(noteBlock.get(INSTRUMENT), noteBlock.get(NOTE), BlockPos(pos))
+                    }
+                }
+            }
+        }
+    }
+
+    private fun loadSong() {
+        mc.player?.let {
+            playingSong = false
+            try {
+                MidiParser.parse("songs${File.separator}$songName").let {
+                    noteSequence = it.first
+                    channelMap = it.second
+                }
+                displayActionbar(it, "Loaded song songs${File.separator}$songName")
+                elapsed = 0
+            } catch (e: Exception) {
+                if (e is IOException)
+                    displayActionbar(it, "Sound not found songs${File.separator}$songName")
+                else
+                    displayActionbar(it, "Invalid MIDI Date: songs${File.separator}$songName")
+            }
```

Comment:
This is not how to handle multiple exception types.
Use multiple `catch` blocks:
```suggestion
            } catch (e: IOException) {
                displayActionbar(it, "Sound not found songs${File.separator}$songName")
            } catch (e: InvalidMidiDataException) {
                displayActionbar(it, "Invalid MIDI Date: songs${File.separator}$songName")
            }
```